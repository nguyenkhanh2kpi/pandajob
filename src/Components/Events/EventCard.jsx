import { ArrowForwardIcon, CopyIcon, StarIcon } from '@chakra-ui/icons'
import { Badge, Box, Button, Card, CardBody, Flex, Heading, HStack, IconButton, Image, SlideFade, Stack, Text, Wrap, WrapItem } from '@chakra-ui/react'
import React from 'react'
import { useNavigate } from 'react-router-dom'
import { ToastContainer, toast } from 'react-toastify'

export const EventCard = ({ id, title, article, time, author, image, content, status }) => {
  const navigate = useNavigate()
  const property = {
    imageUrl: 'https://bit.ly/2Z4KKcF',
  }
  const copyToClipboard = (e) => {
    const textToCopy = `http://localhost:3000/event/${e.target.value}`
    navigator.clipboard
      .writeText(textToCopy)
      .then(() => {
        toast.success('Đã sao chép vào clipboard: ', textToCopy)
      })
      .catch((err) => {
        console.error('Lỗi khi sao chép vào clipboard ', err)
      })
  }
  return (
    <SlideFade in={true} offsetX='-20px'>
      <Box
        backgroundColor='#ffffff'
        maxW='sm'
        borderRadius='lg'
        overflow='hidden'
        fontFamily={'Montserrat'}
        _hover={{
          boxShadow: 'xl',
          transition: 'all 0.2s ease-in-out',
          transform: 'translate(2px, -5px)',
        }}>
        <Image src={image} alt={property.imageUrl} width='400px' height='262px' />

        <Box p='6'>
          <Box display='flex' alignItems='baseline'>
            <Badge borderRadius='full' px='2' colorScheme='teal'>
              New
            </Badge>
            <Box color='gray.500' fontWeight='semibold' letterSpacing='wide' fontSize='xs' textTransform='uppercase' ml='2'>
              {property.beds} beds &bull; {property.baths} baths
            </Box>
          </Box>

          <Box mt='1' fontWeight='semibold' as='h4' lineHeight='tight' noOfLines={1}>
            {title}
          </Box>

          <Box noOfLines={3}>{article}</Box>

          <Box display='flex' mt='2' alignItems='center'>
            <Button rightIcon={<ArrowForwardIcon />} colorScheme='teal' variant='outline' onClick={() => navigate('/event/' + id)} w='200px'>
              View more
            </Button>
            <IconButton value={id} aria-label='Search database' icon={<CopyIcon />} marginLeft='auto' onClick={copyToClipboard} />
          </Box>
        </Box>
      </Box>
    </SlideFade>
  )
}
