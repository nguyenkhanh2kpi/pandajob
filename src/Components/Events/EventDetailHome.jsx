import { Badge, Box, Button, Center, HStack, Heading, IconButton, Image, SlideFade, VStack } from '@chakra-ui/react'
import React, { useEffect, useState } from 'react'
import { useNavigate, useParams } from 'react-router-dom'
import { eventService } from '../../Service/event.service'
import { ArrowForwardIcon, CopyIcon } from '@chakra-ui/icons'

export const EventDetailHome = () => {
  const params = useParams()
  const navigate = useNavigate()
  const [event, setEvent] = useState({
    id: 0,
    title: '',
    article: '',
    time: '',
    status: true,
    image: '',
    content: '',
  })

  useEffect(() => {
    eventService.getEventById(params.id).then((res) => {
      setEvent(res)
    })
  }, [])

  return (
    <VStack fontFamily={'Montserrat'} m={2} p={2}>
      <SlideFade in={true} offsetY={20}>
        <Heading size={'lg'} m={'6'} mt={24} ml={2} textAlign={'left'} marginRight='auto'></Heading>
      </SlideFade>
      <HStack align={'flex-start'} w={'70vw'} alignItems={'center'} display={'flex'} justifyContent={'center'}>
        <Box maxW='960px' borderWidth='1px' borderRadius='lg' overflow='hidden' w={'100%'}>
          <Image src={event.image} alt='Image' />

          <Box p='6'>
            <Box display='flex' mt={2} alignItems='baseline'>
              <Badge borderRadius='full' px='2' colorScheme='teal'>
                Author
              </Badge>
              <Box color='gray.500' fontWeight='semibold' letterSpacing='wide' fontSize='xs' textTransform='uppercase' ml='2'>
                {event.author}
              </Box>
            </Box>
            <Box display='flex' mt={2} alignItems='baseline'>
              <Badge borderRadius='full' px='2' colorScheme='teal'>
                Time
              </Badge>
              <Box color='gray.500' fontWeight='semibold' letterSpacing='wide' fontSize='xs' textTransform='uppercase' ml='2'>
                {event.time}
              </Box>
            </Box>

            <Box mt='1' fontWeight='semibold' as='h4' lineHeight='tight' noOfLines={1}>
              {event.title}
            </Box>

            <Box>{event.article}</Box>
          </Box>
        </Box>
      </HStack>
      <HStack align={'flex-start'} w={'70vw'} alignItems={'center'} display={'flex'} justifyContent={'center'}>
        <Box maxW='960px' borderWidth='1px' borderRadius='lg' overflow='hidden' w={'100vw'}>
          <Box m={5}>{event.content}</Box>
        </Box>
      </HStack>
    </VStack>
  )
}
