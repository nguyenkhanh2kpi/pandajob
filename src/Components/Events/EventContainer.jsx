import React, { useEffect, useState } from 'react'
import { eventService } from '../../Service/event.service'
import { ToastContainer, toast } from 'react-toastify'

import { Center, Heading, SimpleGrid, SlideFade, Spinner, VStack } from '@chakra-ui/react'
import { EventCard } from './EventCard'

export const EventContainer = () => {
  const [events, setEvent] = useState()
  useEffect(() => {
    eventService
      .getEvent()
      .then((res) => setEvent(res))
      .catch((er) => console.log(er))
  })
  return (
    <VStack width={'100vw'} align={'flex-start'} m={2} p={2}>
      {!events ? (
        <Center direction='row' spacing={4} w={'80vw'} h={'20vw'}>
          <Spinner color='blue.500' size='xl' />
        </Center>
      ) : (
        <SimpleGrid columns={3} spacing={5}>
          {events
            .filter((event) => event.status === true)
            .map((event) => (
              <EventCard key={event.id} {...event} />
            ))}
        </SimpleGrid>
      )}

      <ToastContainer
        position='bottom-right'
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        theme='light'
      />
    </VStack>
  )
}
