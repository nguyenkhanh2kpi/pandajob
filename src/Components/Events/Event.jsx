import React from 'react'
import { Heading, HStack, SlideFade, VStack } from '@chakra-ui/react'
import { EventContainer } from './EventContainer'
import { ToastContainer, toast } from 'react-toastify'

export const Events = () => {
  return (
    <VStack fontFamily={'Montserrat'}>
      <SlideFade in={true} offsetY={20}>
        <Heading size={'lg'} m={'6'} mt={24} ml={2} textAlign={'left'} marginRight='auto'></Heading>
      </SlideFade>
      <HStack align={'flex-start'} w={'70vw'}>
        <EventContainer />
      </HStack>
    </VStack>
  )
}
