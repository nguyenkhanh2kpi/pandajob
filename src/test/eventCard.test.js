import React from 'react'
import { render, screen, fireEvent, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { BrowserRouter } from 'react-router-dom'
import { EventCard } from '../Components/Events/EventCard'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}))

jest.mock('react-toastify', () => ({
  toast: {
    success: jest.fn(),
  },
}))

test('renders EventCard with correct props', () => {
  const { getByText } = render(
    <BrowserRouter>
      <EventCard id='1' title='Test Event' article='This is a test article.' time='2024-06-23' author='John Doe' image='https://bit.ly/2Z4KKcF' content='Some content' status='active' />
    </BrowserRouter>
  )

  expect(getByText('Test Event')).toBeInTheDocument()
  expect(getByText('This is a test article.')).toBeInTheDocument()
})

test('navigates to event page on button click', () => {
  const navigate = jest.fn()
  require('react-router-dom').useNavigate.mockReturnValue(navigate)

  const { getByText } = render(
    <BrowserRouter>
      <EventCard id='1' title='Test Event' article='This is a test article.' time='2024-06-23' author='John Doe' image='https://bit.ly/2Z4KKcF' content='Some content' status='active' />
    </BrowserRouter>
  )

  fireEvent.click(getByText('Xem thÃªm'))
  expect(navigate).toHaveBeenCalledWith('/event/1')
})

test('matches snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <EventCard id='1' title='Test Event' article='This is a test article.' time='2024-06-23' author='John Doe' image='https://bit.ly/2Z4KKcF' content='Some content' status='active' />
    </BrowserRouter>
  )

  expect(asFragment()).toMatchSnapshot()
})
