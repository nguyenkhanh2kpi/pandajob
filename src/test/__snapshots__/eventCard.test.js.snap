// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-offset-slide"
    style="opacity: 0; transform: translateX(-20px) translateY(8px) translateZ(0);"
  >
    <div
      class="css-uyjx1l"
    >
      <img
        alt="https://bit.ly/2Z4KKcF"
        class="chakra-image css-amzuzo"
        src="https://bit.ly/2Z4KKcF"
      />
      <div
        class="css-1vae7t2"
      >
        <h4
          class="css-dzd7d1"
        >
          Test Event
        </h4>
        <div
          class="css-zvlevn"
        >
          This is a test article.
        </div>
        <div
          class="css-15yujhg"
        >
          <button
            class="chakra-button css-4cc577"
            name="1"
            type="button"
          >
            Xem thÃªm
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Search database"
            class="chakra-button css-12we9zr"
            type="button"
            value="1"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
